app/controllers/groups_controller.rb

```diff
@@ -11,15 +11,22 @@ def new
 
   def create
     @group = Group.new(group_params)
-    @group.user_ids= current_user.id
+    @group.user_ids = current_user.id # Add user as a user
+    @group.owner_id = current_user.id # Add user as owner
     if @group.save
-      redirect_to dashboard_path
+      create_list
+      redirect_to group_path(@group)
     else
-      render 'new'
+      flash[:notice] = 'The group name is already taken. Please choose another name.'
+      redirect_to new_group_path
```

Comment:
curious why you're using a redirect here instead of a render? I saw you used render in most of your else blocks in other controllers

Reply:
I'm not sure, but I've changed it to `render 'new'` now to preserve the user's input.